src/components/AxisCollection/index.js

```diff
@@ -47,45 +50,93 @@ class AxisCollection extends React.Component {
   onAxisMouseLeave = seriesId =>
     this.props.onMouseLeave ? e => this.props.onMouseLeave(e, seriesId) : null;
 
+  getAxisOffsets = () => {
+    const { series, yAxisPlacement, yAxisWidth } = this.props;
+
+    const numCollapsed = series.filter(
+      this.axisFilter(AxisDisplayMode.COLLAPSED)
+    ).length;
+    const numVisible = series.filter(this.axisFilter(AxisDisplayMode.ALL))
+      .length;
+
+    switch (yAxisPlacement) {
+      case AxisPlacement.LEFT:
+        return {
+          collapsed: 0,
+          visible: numCollapsed ? yAxisWidth : 0,
+        };
+      case AxisPlacement.BOTH:
+        throw new Error(
+          'BOTH is not a valid option for AxisCollection -- please specify RIGHT or LEFT'
+        );
+      case AxisPlacement.RIGHT:
+      case AxisPlacement.UNSPECIFIED:
+      default:
+        return {
+          collapsed: numVisible * yAxisWidth,
+          visible: 0,
+        };
+    }
+  };
+
   axisFilter = mode => s =>
     !s.hidden && (s.yAxisDisplayMode || this.props.axisDisplayMode) === mode;
 
-  renderAllVisibleAxes() {
+  placementFilter = s =>
+    !s.yAxisPlacement ||
+    s.yAxisPlacement === AxisPlacement.BOTH ||
+    s.yAxisPlacement === this.props.yAxisPlacement;
+
+  renderAllVisibleAxes = offsetx => {
     const {
       collections,
       series,
       zoomable,
       height,
       updateYTransformation,
+      yAxisPlacement,
       yAxisWidth,
       yTransformations,
     } = this.props;
-    let axisOffsetX = -yAxisWidth;
+    let axisOffsetX = offsetx - yAxisWidth;
+
+    const filteredSeries = series
+      .filter(this.axisFilter(AxisDisplayMode.ALL))
+      .filter(this.placementFilter);
+    if (yAxisPlacement === AxisPlacement.LEFT) {
+      filteredSeries.reverse();
+    }
+
+    const filteredCollections = collections
+      .filter(this.axisFilter(AxisDisplayMode.ALL))
+      .filter(this.placementFilter);
+    if (yAxisPlacement === AxisPlacement.LEFT) {
+      filteredCollections.reverse();
+    }
+
     return []
       .concat(
-        series
-          .filter(s => !s.collectionId)
-          .filter(this.axisFilter(AxisDisplayMode.ALL))
-          .map(s => {
-            axisOffsetX += yAxisWidth;
-            return (
-              <YAxis
-                key={`y-axis--${s.id}`}
-                offsetx={axisOffsetX}
-                zoomable={s.zoomable !== undefined ? s.zoomable : zoomable}
-                series={s}
-                height={height}
-                width={yAxisWidth}
-                updateYTransformation={updateYTransformation}
-                yTransformation={yTransformations[s.id]}
-                onMouseEnter={this.onAxisMouseEnter(s.id)}
-                onMouseLeave={this.onAxisMouseLeave(s.id)}
-              />
-            );
-          })
+        filteredSeries.filter(s => s.collectionId === undefined).map(s => {
+          axisOffsetX += yAxisWidth;
```

Comment:
Will this push the first axis `yAxisWidth` pixels in as well?

Reply:
Yeah, but note that on L101 I change it to subtract the yAxisWidth to start.